src/main/java/pl/trainingapp/controller/ExerciseController.java

```diff
@@ -1,13 +1,48 @@
 package pl.trainingapp.controller;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 import pl.trainingapp.services.ExerciseService;
 
+
 @RestController
 @RequiredArgsConstructor
 public class ExerciseController {
 
     private final ExerciseService exerciseService;
 
+    @GetMapping("/exercises")
+    public String getAllExercises() {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            return objectMapper.writeValueAsString(exerciseService.getAllExercises());
```

Comment:
You don't need to serialize responses in controller, jackson library that you already have will handle that. Just return object or list of objects

Reply:
done!